File: r/globalopt/examples/benchmark_example.R
library(globalopt)

out_dir <- file.path("r", "globalopt", "examples", "output")
dir.create(out_dir, recursive = TRUE, showWarnings = FALSE)

tbl <- run_benchmarks(
  dimensions = c(2L, 10L),
  budgets = c(1000L),
  seeds = 1:5
)

csv_path <- file.path(out_dir, "benchmark_results.csv")
md_path <- file.path(out_dir, "benchmark_results.md")

write_benchmark_csv(tbl, csv_path)
write_benchmark_markdown(tbl, md_path)

cat("wrote", csv_path, "\n")
cat("wrote", md_path, "\n")
print(head(tbl, 12L))

File: python/examples/benchmark_example.py
from pathlib import Path

from globalopt.benchmarks import (
    benchmark_table,
    random_search_optimizer,
    run_benchmarks,
    write_benchmark_csv,
    write_benchmark_markdown,
)


def main() -> None:
    try:
        summaries = run_benchmarks(
            dimensions=(2, 10),
            budgets=(1000,),
            seeds=tuple(range(1, 6)),
            optimizers={"random_search": random_search_optimizer},
            include_scipy_de=True,
        )
    except Exception:
        summaries = run_benchmarks(
            dimensions=(2, 10),
            budgets=(1000,),
            seeds=tuple(range(1, 6)),
            optimizers={"random_search": random_search_optimizer},
            include_scipy_de=False,
        )
    out_dir = Path("python/examples/output")
    csv_path = write_benchmark_csv(out_dir / "benchmark_results.csv", summaries)
    md_path = write_benchmark_markdown(out_dir / "benchmark_results.md", summaries)
    print(f"wrote {csv_path}")
    print(f"wrote {md_path}")
    for row in benchmark_table(summaries)[:12]:
        print(row)


if __name__ == "__main__":
    main()
